### PR TITLE
[AIRFLOW-2009] Fix dataflow hook connection-id

### PR DESCRIPTION
**Pull request by fenglu-db**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2009
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Fix DataflowJavaOperator and DataflowPythonOperator to use the provided gcp_conn_id when launching dataflow jobs.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests.contrib.hooks.test_gcp_api_base_hook.test_provide_gcp_credential_file_decorator_key_content
tests.contrib.hooks.test_gcp_api_base_hook.test_provide_gcp_credential_file_decorator_key_path

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by fenglu-db** [comment]
@kaxil @Fokko PTAL, thank you.

File: airflow/contrib/hooks/gcp_api_base_hook.py [hunk 1]
```diff
@@ -24,13 +24,16 @@
...
+import os
+import tempfile
 
 from airflow.exceptions import AirflowException
 from airflow.hooks.base_hook import BaseHook
 from airflow.utils.log.logging_mixin import LoggingMixin
 
 
 _DEFAULT_SCOPES = ('https://www.googleapis.com/auth/cloud-platform',)
+G_APP_CRED = "GOOGLE_APPLICATION_CREDENTIALS"
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_api_base_hook.py]: Can you change a constant name? The current name is an abbreviation of the value, but it does not describe its purpose.

Comment (fenglu-db) [reply airflow/contrib/hooks/gcp_api_base_hook.py]: Done. On a second thought, I am going to make this private too as it's only used in gcp_api_base_hook.

**Review by sdevani — APPROVED** [review]
Looks good to me.

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Thanks @fenglu-g

**Pull request merged**